packages/openapi-to-graphql/src/utils.ts

```diff
@@ -51,6 +51,192 @@ export const mitigations = {
+  return (
+    typeof n === 'number' &&
+    isFinite(n) &&
+    n <= MAX_LONG &&
+    n >= MIN_LONG &&
+    n % 1 === 0
+  )
+}
+
+/**
+ *
+ */
+
+export function isSafeFloat(n: unknown): boolean {
+  return typeof n === 'number' && n % 0.5 !== 0
+}
+
+/**
+ * convert a date and/or date-time string into a date object
+ */
+
+function toDate(n: string) {
+  const parsed = Date.parse(n)
+  const $ref = new Date()
+
+  $ref.setTime(parsed)
+
+  return (
+    (typeof parsed === 'number' &&
+      parsed !== NaN &&
+      parsed > 0 &&
+      String(parsed).length === 13 &&
+      $ref) ||
+    null
+  )
+}
+
+/**
+ * serialize a date string into the ISO format
+ */
+
+export function serializeDate(n: string) {
+  const date = toDate(n)
+  return date && date.toJSON()
+}
+
+/**
+ * verify that a vriable contains a safe date/date-time string
+ */
+
+export function isSafeDate(n: string): boolean {
+  const date = toDate(n)
+  return date !== null && date.getTime() !== NaN
+}
+
+/**
+ * verify is a string is a valid URL
+ */
+
+export function isURL(s: string): boolean {
+  let res = null
+  const urlRegex = /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z0-9]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g
+  try {
+    res = s.match(urlRegex)
+  } catch (e) {
+    res = null
+  }
+  return res !== null
+}
+
+/**
+ * verify if a string is a valid EMAIL
+ * See: https://github.com/Urigo/graphql-scalars/blob/master/src/resolvers/EmailAddress.ts#L4
+ */
+
+export function isEmail(s: string): boolean {
+  const emailRegex = /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+  return emailRegex.test(s)
+}
+
+/**
+ * verify if a string is a valid GUID/UUID
+ * See: https://github.com/Urigo/graphql-scalars/blob/master/src/resolvers/GUID.ts#L4
+ */
+
+export function isUUIDOrGUID(s: string): boolean {
+  const uuidRegExp = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+  const guidRegExp = /^(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}$/gi
+
+  if (s.startsWith('{')) {
+    s = s.substring(1, s.length - 1)
+  }
+
+  return uuidRegExp.test(s) || guidRegExp.test(s)
+}
+
+/**
+ * convert the fist letter of a word in a string to upper case
+ */
+
+export function ucFirst(s: string): string {
+  if (typeof s !== 'string') {
+    return ''
+  }
+
+  return s.replace(/^./, c => c.toUpperCase())
+}
+
+/**
+ * check if a literal is falsy or not
+ */
+const isLiteralFalsey = (variable: unknown): boolean => {
+  return variable === '' || variable === false || variable === 0
+}
+
+/**
+ * check if a variable contains a reference type (not a literal)
+ */
+
+const isPrimitive = (arg: any): boolean => {
+  return (
+    typeof arg === 'object' || (Boolean(arg) && typeof arg.apply === 'function')
+  )
+}
+
+/**
+ * provide the name of primitive and/or reference types
+ */
+const checkTypeName = (target: unknown, type: string): boolean => {
```

Comment:
@isocroft Not sure the above comment describes what this function does accurately. It seems again to check that the given target (why not use "value" here?) is of the given type. Is that correct? The comment makes it seem as if a name of a primitive and/or type is returned.

Reply:
@ErikWittern I have modified the comment to reflect the intent of the fucntion.